File: libraries/clib/net/SocketOutputStream.c
/*
 * java.net.SocketOutputStream.c
 *
 * Copyright (c) 1996, 1997
 *	Transvirtual Technologies, Inc.  All rights reserved.
 *
 * See the file "lib-license.terms" for information on usage and redistribution 
 * of this file. 
 */

#include "config.h"
#include "config-std.h"
#include "config-mem.h"
#include "config-io.h"
#include <native.h>
#include "../native/FileDescriptor.h"
#include "../native/FileOutputStream.h"
#include "SocketImpl.h"
#include "SocketOutputStream.h"
#include "nets.h"
#include <jsyscall.h>

void
java_net_SocketOutputStream_socketWrite(struct Hjava_net_SocketOutputStream* this, HArrayOfByte* buf, jint offset, jint len)
{
	int r;

	if (unhand(unhand(unhand(this)->impl)->fd)->fd < 0) {
		return;
	}
	r = sockwrite(unhand(unhand(unhand(this)->impl)->fd)->fd, &unhand(buf)->body[offset], len);
	if (r < 0) {
		SignalError("java.io.IOException", SYS_ERROR);
	}
}

File: libraries/javalib/java/net/SocketOutputStream.java
package java.net;

import java.io.OutputStream;
import java.io.IOException;

/*
 * Java core library component.
 *
 * Copyright (c) 1997, 1998
 *      Transvirtual Technologies, Inc.  All rights reserved.
 *
 * See the file "license.terms" for information on usage and redistribution
 * of this file.
 */
class SocketOutputStream
  extends OutputStream {

private SocketImpl impl;
private byte[] buf = new byte[1];

public SocketOutputStream(SocketImpl impl) {
	this.impl = impl;
}

public void write(byte b[]) throws IOException {
	write(b, 0, b.length);
}

public void write(byte b[], int off, int len) throws IOException {
	impl.write(b, off, len);
}

public void write(int b) throws IOException {
	buf[0] = (byte)b;
	write(buf);
}

public void close() throws IOException {
	impl.close();
}

}
